src/bounce_solution/bounce_solution.cpp

```diff
@@ -0,0 +1,1256 @@
+                  "Completion temperature T = " + std::to_string(trans_temp) +
+                      " chosen as transition temperature.");
+  }
+  return trans_temp;
+}
+
+/**
+ * @brief GetPTStrength Get PT strength alpha
+ */
+double BounceSolution::GetPTStrength() const
+{
+  return alpha;
+}
+
+/**
+ * @brief Storage of the tunneling rate per volume of the transition from
+ * false to true vacuum
+ * @param Temp temperature
+ */
+double BounceSolution::Tunneling_Rate(const double &Temp)
+{
+  if (Temp < SolutionList.front().T or Temp > SolutionList.back().T)
+    return 0; // Never extrapolate
+  double Shat3 = GetBounceSol(Temp);
+  if (Shat3 < 0)
+  {
+    Logger::Write(LoggingLevel::BounceDetailed,
+                  "Action spline became negative somewhere. T = " +
+                      std::to_string(Temp));
+    return 1e100; // Spline is unstable or ridiculous action. Return values that
+                  // kills the tunneling rate
+  }
+
+  double amp =
+      std::pow(Temp, 4) * std::pow((Shat3 / (2 * M_PI * Temp)), 3. / 2);
+  return amp * std::exp(-Shat3 / Temp);
+}
+/**
+ * @brief Storage of the temperature-dependent Hubble rate
+ * @param Temp temperature
+ */
+double BounceSolution::Hubble_Rate(const double &Temp)
+{
+  return M_PI * std::sqrt(this->GetGstar()) / std::sqrt(90.) *
+         std::pow(Temp, 2) / MPl; // radiation dominated universe
+}
+
+/**
+ * @brief CalcGstar Calculate the number of effective degrees of freedom as a
+ * function of temperature, higgs masses are evaluated at global minimum
+ * @param Temp temperature
+ */
+void BounceSolution::CalcGstar(const double &Temp)
+{
+  std::size_t NHiggs = this->modelPointer->get_NHiggs();
+
+  const std::vector<double> HiggsMassesSquared =
+      this->modelPointer->HiggsMassesSquared(
+          this->modelPointer->MinimizeOrderVEV(
+              this->modelPointer->get_vevTreeMin()),
+          Temp); // todo: calculate HiggsMassesSquared not in global, but in
+                 // minimum in which we are still in (e.g. local one until
+                 // T < Tperc)
+
+  std::size_t HiggsAboveThreshold = 0;
+
+  for (double mass : HiggsMassesSquared)
+  {
+    if (mass > Temp)
+    {
+      HiggsAboveThreshold++;
+    }
+  }
+
+  double gb   = 8 * 2 + 3 * 3 + 2 + NHiggs - HiggsAboveThreshold;
+  double gf   = 6 * 3 * 2 * 2 + 3 * 2 * 2 + 3 * 2;
+  double geff = gb + 7. / 8 * gf;
+
+  this->SetGstar(geff);
+}
+
+/**
+ * @brief CalcGstarPureRad Calculate the number of effective degrees of
+ * freedom assuming a purely radiative universe
+ */
+void BounceSolution::CalcGstarPureRad()
+{
+  std::size_t NHiggs = this->modelPointer->get_NHiggs();
+
+  double gb   = 8 * 2 + 3 * 3 + 2 + NHiggs;
+  double gf   = 6 * 3 * 2 * 2 + 3 * 2 * 2 + 3 * 2;
+  double geff = gb + 7. / 8 * gf;
+
+  this->SetGstar(geff);
+}
+
+/**
+ * @brief Calculation of nucleation temperature
+ */
+void BounceSolution::CalculateNucleationTemp()
+{
+  if (status_bounce_sol == "success")
```

Comment:
This is why the status should be an enum. Easy to make a typo here

Reply:
Done.